### PR TITLE
Adds logging for #105122

### PR DESCRIPTION
**Pull request by benwtrent**
related to: https://github.com/elastic/elasticsearch/issues/105122

### CODE REVIEW

File: server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/IndexRecoveryIT.java [hunk 1]
```diff
@@ -1039,7 +1039,10 @@ public void testHistoryRetention() throws Exception {
         assertThat(recoveryState.getTranslog().recoveredOperations(), greaterThan(0));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/105122")
+    @TestLogging(
+        reason = "https://github.com/elastic/elasticsearch/issues/105122",
+        value = "org.elasticsearch.indices.recovery:TRACE, org.elasticsearch.index.shard:TRACE, org.elasticsearch.index.engine:TRACE"
```
Comment (DaveCTurner) [inline server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/IndexRecoveryIT.java:1044]: I think the spaces will prevent this working:

```suggestion
        value = "org.elasticsearch.indices.recovery:TRACE,org.elasticsearch.index.shard:TRACE,org.elasticsearch.index.engine:TRACE"
```

File: server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/IndexRecoveryIT.java [hunk 2]
```diff
@@ -1039,7 +1039,10 @@ public void testHistoryRetention() throws Exception {
         assertThat(recoveryState.getTranslog().recoveredOperations(), greaterThan(0));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/105122")
+    @TestLogging(
+        reason = "https://github.com/elastic/elasticsearch/issues/105122",
```
Comment (DaveCTurner) [inline server/src/internalClusterTest/java/org/elasticsearch/indices/recovery/IndexRecoveryIT.java:1043]: I think `TestIssueLogging` is the better annotation in this case (although it's not functionally different from `TestLogging`.

```suggestion
    @TestIssueLogging(
        issueUrl = "https://github.com/elastic/elasticsearch/issues/105122",
```

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
I got it to fail locally with verbose logging. Closing this for now.

**Pull request closed without merging**